collectors/python.d.plugin/dns_query_time/dns_query_time.chart.py

```diff
@@ -90,10 +90,13 @@ def dns_req(ns, t, q):
 
         try:
             dns_start = time()
-            dns.query.udp(request, ns, timeout=t)
+            r = dns.query.udp(request, ns, timeout=t)
             dns_end = time()
-            query_time = round((dns_end - dns_start) * 1000)
-            q.put({'_'.join(['ns', ns.replace('.', '_')]): query_time})
+            if not r.answer:
```

Comment:
i suggest to check `rcode` 
http://www.dnspython.org/docs/1.14.0/dns.rcode-module.html
```python
resp = dns.query.udp(request, ns, timeout=t)
if not (resp and resp.rcode == dns.rcode.NOERROR):
    put
    return
```

Reply:
> i suggest also to check `rcode`
> http://www.dnspython.org/docs/1.14.0/dns.rcode-module.html
> 
> ```python
> resp = dns.query.udp(request, ns, timeout=t)
> if not (resp and resp.rcode == dns.rcode.NOERROR):
>     put
>     return
> ```

This doesn't seem to work, in my case resp will exist but not resp.answer:
```NO RESPONSE: id 63190
opcode QUERY
rcode NOERROR
flags QR RD RA
;QUESTION
sub.domain.com. IN A
;ANSWER
;AUTHORITY
domain.com. 445 IN SOA a1.verisigndns.com. mdnshelp.verisign.com. 1533674417 28800 7200 1209600 86400
;ADDITIONAL```
note rcode is NOERROR but no data exists in answer section, it's not a common failure scenario I don't think, but it's the one I'm battling at the moment.

> btw all these '-100', '-50' are really wrong, it was a bad choice to use negative number to show errors..
Would it be possible to return 'None' instead of -100? It seems to break the code trying it with debug mode.

Reply:
oh,you are right, i reviewed at night :see_no_evil: 	

```python
if not (resp.rcode == dns.rcode.NOERROR and resp.answer):
```

btw this is source code

http://www.dnspython.org/docs/1.15.0/dns.query-pysrc.html#udp

i see response has `time` attribute, lets use it!

```python
250      finally: 
251          if begin_time is None: 
252              response_time = 0 
253          else: 
254              response_time = time.time() - begin_time 
255          s.close() 
256      r = dns.message.from_wire(wire, keyring=q.keyring, request_mac=q.mac, 
257                                one_rr_per_rrset=one_rr_per_rrset) 
258      r.time = response_time 
```

Reply:
> Would it be possible to return a null value instead of -100?

It is very easy to do, the problem that it breaks alarms, which is not acceptable.

Reply:
the response_time value is in seconds, multiplying by 1000 to get ms and rounding gives many responses of `0` - removing rounding still produces an integer, is zero an acceptable response?

Reply:
I guess it is < 0.5 ms that is why it is rounded to 0.

to make it more precise multiply `query_time` by 100000 in dns_request
and

change 
https://github.com/netdata/netdata/blob/90108a57d15f6714669daa61ba8e7603d7da1218/collectors/python.d.plugin/dns_query_time/dns_query_time.chart.py#L146-L148

to 
```python
 '_'.join(['ns', ns.replace('.', '_')]), 
 ns, 
 'absolute', 
 1,
 100,
```